frontend/src/components/SemesterBlock.tsx

```diff
@@ -74,19 +77,71 @@ interface ReduxDispatchSemesterBlockProps {
 
 interface SemesterBlockProps {
   semester: DNDScheduleTerm;
+  isEditable: boolean;
 }
 
 type Props = SemesterBlockProps &
   ReduxStoreSemesterBlockProps &
   ReduxDispatchSemesterBlockProps;
 
+type NonEditProps = SemesterBlockProps & ReduxStoreSemesterBlockProps;
+
 interface SemesterBlockState {
   modalVisible: boolean;
   snackbarOpen: boolean;
   deletedClass?: DNDScheduleCourse;
 }
 
-class SemesterBlockComponent extends React.Component<
+function renderTooltip(warnings: IWarning[]) {
+  return (
+    <div style={{ display: "flex", flexDirection: "column" }}>
+      {warnings.map((w, index) => {
+        return <span key={index}>{w.message}</span>;
+      })}
+    </div>
+  );
+}
+
+const NonEditableSemesterBlockComponent = (props: NonEditProps) => {
+  const renderBody = (props: NonEditProps) => {
+    const { semester, courseWarnings } = props;
+    const status = semester.status;
+    return semester.classes.map((scheduleCourse, index) => {
+      if (!!scheduleCourse) {
+        return (
+          <NonDraggableClassBlock
+            key={index}
+            course={scheduleCourse}
+            warnings={findCourseWarnings(courseWarnings, scheduleCourse)}
+            onDelete={() => {}}
+            hideDelete={true}
+          />
+        );
+      } else return <EmptyBlock key={index} />;
+    });
+  };
+  return (
+    <OutsideContainer>
+      <Container warning={false}>
+        <ClassListWrapper>
+          {props.warnings.length > 0 ? (
+            <Tooltip
+              title={renderTooltip(props.warnings)}
+              placement="top"
+              arrow
+            >
+              <>{renderBody(props)}</>
```

Comment:
Do we need the `<>`?

Reply:
Unfortunately yeah, it's because renderBody returns a list and tooltip can only have one child